### PR TITLE
Add validator

### PR DESCRIPTION
**Pull request by karthikeyan-cere**
Add validator

### CODE REVIEW

File: scripts/tokenomics-emulation/config.json [hunk 1]
```diff
@@ -12,10 +22,12 @@
     "genesis_validators_stake": 10000,
     "genesis_councils_amount": 1,
     "genesis_councils_stake": 100000,
+    "sync_wait_time": 15,
     "validators":{
       "amount": 3,
       "stash_stake": 10000,
-      "controller_stake": 10000
+      "controller_stake": 10000,
+      "commission": 10000
```
Comment (AndreiNavoichyk) [inline scripts/tokenomics-emulation/config.json:30]: Here should be value from 0 to 100.

File: scripts/tokenomics-emulation/config.json [hunk 2]
```diff
@@ -1,5 +1,15 @@
 {
   "network": {
+    "hosts": [
+      {
```
Comment (AndreiNavoichyk) [inline scripts/tokenomics-emulation/config.json:4]: Here will be all hosts including genesis. Please, add placeholders.

File: scripts/tokenomics-emulation/emulations/validators.emulation.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+class ValidatorsEmulation implements IEmulation {
+  constructor(
+    private readonly config,
+    private readonly account: Accounts
+  ) {}
+
+  public async run(): Promise<void> {
+    const genericValidator = this.config.network.validators.amount;
+    const commissionValue = this.config.network.nominators.commission;
+    const stashBond = this.config.network.nominators.stash_stake;
```
Comment (AndreiNavoichyk) [inline scripts/tokenomics-emulation/emulations/validators.emulation.ts:16]: It should be `stash_stake` minus the existential deposit.

File: scripts/tokenomics-emulation/index.ts [hunk 4]
```diff
@@ -88,6 +89,8 @@ async function main() {
     new EmulationsFactory(network, account, ddcContract, cereContract, batcher),
   );
   await emulations.run();
+  const validator = new ValidatorsEmulation(config, account);
```
Comment (AndreiNavoichyk) [inline scripts/tokenomics-emulation/index.ts:92]: Include it in the common emulations flow and add a section in `config.json`.

File: scripts/tokenomics-emulation/validator.ts [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+    console.log(
+      `Controller Account is ${this.controllerAccount.address} and balance is ${this.controllerBalance}\n`
+    );
+  }
+
+  /**
+   * Generate session key
+   */
+  public async generateSessionKey() {
+    console.log(`\nGenerating Session Key`);
```
Comment (AndreiNavoichyk) [inline scripts/tokenomics-emulation/validator.ts:61]: Add `\n` in the end. Here and everywhere.

File: scripts/tokenomics-emulation/config.json (hunk 1 above)
Comment (karthikeyan-cere) [reply scripts/tokenomics-emulation/config.json:30]: Changed to 5

File: scripts/tokenomics-emulation/emulations/validators.emulation.ts (hunk 3 above)
Comment (karthikeyan-cere) [reply scripts/tokenomics-emulation/emulations/validators.emulation.ts]: added calculation to substrate exsistential deposit

File: scripts/tokenomics-emulation/index.ts (hunk 4 above)
Comment (karthikeyan-cere) [reply scripts/tokenomics-emulation/index.ts]: added

File: scripts/tokenomics-emulation/config.json (hunk 2 above)
Comment (karthikeyan-cere) [reply scripts/tokenomics-emulation/config.json:4]: added

File: scripts/tokenomics-emulation/emulations/validators.emulation.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import { IEmulation } from "./emulation.interface";
+import Network from "../network";
+import Accounts from "../accounts";
+import Validator from "../validator";
+import fs from "fs";
+
+class ValidatorsEmulation implements IEmulation {
```
Comment (AndreiNavoichyk) [inline scripts/tokenomics-emulation/emulations/validators.emulation.ts:7]: Let's rename to `AddValidatorsEmulation`

**Review by AndreiNavoichyk — APPROVED** [review]

File: scripts/tokenomics-emulation/emulations/validators.emulation.ts (hunk 6 above)
Comment (karthikeyan-cere) [reply scripts/tokenomics-emulation/emulations/validators.emulation.ts]: Renamed

**Pull request merged**